src/Razor/src/Microsoft.AspNetCore.Razor.VSCode/src/RazorCSharpLanguageMiddleware.ts

```diff
@@ -69,10 +69,33 @@ export class RazorCSharpLanguageMiddleware implements LanguageMiddleware {
                         continue;
                     }
                 } else {
+                    // Similar to above, this is kind of wrong. We're manually trimming whitespace and adjusting
```

Comment:
❤️ the comment

Reply:
This whole logic needs to move to the codeActionProvider. We can't trim all edits in the LanguageMiddleware layer as this might have unintended side effects. If there are other places that need trimming, we'll need to do it on a case-by-case basis.

Reply:
This should probably move [here](https://github.com/dotnet/aspnetcore-tooling/blob/master/src/Razor/src/Microsoft.AspNetCore.Razor.VSCode/src/CodeActions/RazorFullyQualifiedCodeActionTranslator.ts#L17) to be precise. @ryanbrandenburg can you confirm?

Reply:
It sounds like all this is specific to the behavior of RazorFullyQualifiedCodeActionTranslator, so yes it should go where @ajaybhargavb indicated.

Reply:
Updated logic to route this to the code action translator.